### ISSUE TITLE
Unable to use module on Apply M1. 

### ISSUE THREAD

**Issue by burninmedia** [terraform-google-modules/terraform-google-vault#144]
### TL;DR

│ Error: Incompatible provider version
│
│ Provider registry.terraform.io/hashicorp/tls v2.2.0 does not have a package
│ available for your current platform, linux_arm64.
│
│ Provider releases are separate from Terraform CLI releases, so not all
│ providers are available for all platforms. Other versions of this provider
│ may have different platforms supported.

### Expected behavior

To download a TLS module that is compatible with the latest OSX OS/Arch.

### Observed behavior

Failed to find tls v2.2.0 for linux _arm64

### Terraform Configuration

```hcl
module "vault" {
  source                = "terraform-google-modules/vault/google"
  project_id            = var.project_id
  region                = var.region
  kms_keyring           = var.kms_crypto_key
  kms_crypto_key        = var.kms_crypto_key
  vault_min_num_servers = var.vault_min_num_servers
  vault_allowed_cidrs   = var.vault_allowed_cidrs
  service_label         = var.service_label
  load_balancing_scheme = var.load_balancing_scheme
}
```

### Terraform Version

```sh
Terraform v1.1.8
on linux_arm64
```

### Additional information

This may be an issue resolved with TLS 3.3.X

**Comment by burninmedia** [comment]
#145 Suggested fix for this bug. Update TLS to v3.3 and remove deprecated inputs from the `crypto.tf` file.

**Comment by burninmedia** [comment]
#146 
 The suggested fix for this bug. Update TLS to v3.3 and remove deprecated inputs from the crypto.tf file and adds missing network inputs.
